kite-data/kite-data-core/src/main/java/org/kitesdk/data/ReadiableView.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015 Cloudera.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kitesdk.data;
+
+/**
+ * Readiable views may signal consumers when their underlying data is ready for
+ * consumption. Not all View implementations provide this capability.
+ */
+public interface ReadiableView<E> extends View<E> {
```

Comment:
Updated to Signalable here: 253cbae

I think having the future compatible change to include a ReadySignalable (when other potentially exclusive types of signals are added) makes it a good decision.